conans/test/unittests/client/tools/scm/test_git.py

```diff
@@ -74,6 +89,57 @@ def test_clone_git(self):
         git.clone(path)
         self.assertTrue(os.path.exists(os.path.join(tmp, "myfile")))
 
+    @parameterized.expand([(None,),  # default
+                           ("develop",),  # branch name
+                           ("1.0",),  # tag name
+                           ("HEAD",),  # expression
+                           ])
+    def test_clone_git_shallow(self, element):
+        path, revision = create_local_git_repo({"myfile": "contents"}, commits=3, tags=["1.0"], branch="develop")
+        tmp = temp_folder()
+        git = Git(tmp)
+        git.clone("file://" + path, branch=element, shallow=True)  # --depth is ignored in local clones
+        with self.assertRaises(subprocess.CalledProcessError):
+            git.checkout(element="HEAD~1")
+        self.assertTrue(os.path.exists(os.path.join(tmp, "myfile")))
+        self.assertEqual(git.get_revision(), revision)
+        self.assertEqual(git.run("rev-list --all --count"), "1")
+
+    def test_clone_git_shallow_revision(self):
+        path, revision = create_local_git_repo({"myfile": "contents"}, commits=3, tags=["1.0"], branch="develop")
+        tmp = temp_folder()
+        git = Git(tmp)
+        if Git.get_version() < "2.13":
+            # older Git versions have known bugs with "git fetch origin <sha>":
+            # https://github.com/git/git/blob/master/Documentation/RelNotes/2.13.0.txt
+            #  * "git fetch" that requests a commit by object name, when the other
+            #    side does not allow such an request, failed without much
+            #    explanation.
+            # https://github.com/git/git/blob/master/Documentation/RelNotes/2.14.0.txt
+            # * There is no good reason why "git fetch $there $sha1" should fail
+            #    when the $sha1 names an object at the tip of an advertised ref,
+            #    even when the other side hasn't enabled allowTipSHA1InWant.
+            with self.assertRaises(subprocess.CalledProcessError):
+                git.clone("file://" + path, branch=revision, shallow=True)
+                self.assertEqual(git.get_revision(), revision)
+                self.assertEqual(git.run("rev-list --all --count"), "1")
+        else:
+            git.clone("file://" + path, branch=revision, shallow=True)
+            with self.assertRaises(subprocess.CalledProcessError):
+                git.checkout(element="HEAD~1")
+            self.assertTrue(os.path.exists(os.path.join(tmp, "myfile")))
+
+    def test_clone_git_shallow_with_local(self):
+        path, revision = create_local_git_repo({"myfile": "contents"}, commits=3, tags=["1.0"])
+        tmp = temp_folder()
+        save(os.path.join(tmp, "localfile"), "contents")
+        git = Git(tmp)
```

Comment:
I would add the `localfile` to the repo:

```suggestion
        git = Git(tmp)
        git.run("add localfile")
```

Reply:
but that would be a different case, right? may be another test?

Reply:
if you want to check an untracked file too, just add another one in this test, I think it would be enough

Reply:
okay

Reply:
added